orchagent/portsorch.cpp

```diff
@@ -871,34 +872,60 @@ bool PortsOrch::setPortPfc(sai_object_id_t portId, uint8_t pfc_bitmask)
         return false;
     }
 
-    if (p.m_pfc_asym == SAI_PORT_PRIORITY_FLOW_CONTROL_MODE_COMBINED)
+    if (p.m_pfc_info.pfc_mode == SAI_PORT_PRIORITY_FLOW_CONTROL_MODE_COMBINED)
     {
+        if (pfc_tx_bitmask != pfc_rx_bitmask)
+        {
+            SWSS_LOG_ERROR("Cannot set unequal Tx 0x%x and Rx 0x%x PFC bitmasks "
+                "for the port 0x%" PRIx64 " with disabled \"asymmetric\" PFC mode",
+                pfc_tx_bitmask, pfc_rx_bitmask, portId);
+            return false;
+        }
+
         attr.id = SAI_PORT_ATTR_PRIORITY_FLOW_CONTROL;
+        attr.value.u8 = pfc_tx_bitmask;
+
+        sai_status_t status = sai_port_api->set_port_attribute(portId, &attr);
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("Failed to set PFC bitmask 0x%x for the port 0x%" PRIx64 " (rc:%d)",
```

Comment:
not sure i understand this part.
you are using the tx pfc bitmask for setting flow control and the message in the log is still pfc bitmask which i can see later on you have one for rx and tx.

Reply:
This logic is explained in the "Asymmentric PFC" HLD in the "2.2.1 PortsOrch" section
https://github.com/Azure/SONiC/wiki/Asymmetric-PFC-High-Level-Design#221-portsorch